src/Tools/DynamoShapeManager/Preloader.cs

```diff
@@ -134,6 +134,33 @@ public Preloader(string rootFolder, string shapeManagerPath, LibraryVersion vers
                 Utilities.GeometryFactoryAssembly);
         }
 
+        /// <summary>
+        /// Construct a Preloader by specifying a required library version. If the 
+        /// specified version is invalid, then fallback to construction using
+        /// the standard list of versions.
+        /// </summary>
+        /// <param name="rootFolder">Full path of the directory that contains 
+        /// LibG_xxx folder, where 'xxx' represents the library version. In a 
+        /// typical setup this would be the same directory that contains Dynamo 
+        /// core modules. This must represent a valid directory.
+        /// </param>
+        /// <param name="version">A string representing the version of the shape manager.</param>
+        /// <returns></returns>
+        public static Preloader FromSpecifiedLibraryVersionString(string rootFolder, string version)
+        {
+            try
+            {
+                LibraryVersion libVersion;
+                return LibraryVersion.TryParse(version, out libVersion) ? 
+                    new Preloader(rootFolder, new[] { libVersion }) : 
+                    new Preloader(rootFolder);
+            }
+            catch (Exception ex)
+            {
+                throw new Exception("The specified string value did not map to a supported version of the shape manager.");
+            }
+        }
+
```

Comment:
From an API consistency standpoint I would like this method updated, let's take strongly typed `LibraryVersion` as second argument instead of `string` (consequently removing the need of that exception).

The job of parsing `LibraryVersion` should be moved into `RemoteTestSessionConfig`, since (unlike `Revit add-ons`) `TestServices` can have build-time dependency on `DynamoShapeManager.dll` (they are both in the same folder).